src/main/java/org/sagebionetworks/bridge/services/EnrollmentService.java

```diff
@@ -109,8 +113,10 @@ protected DateTime getWithdrawalDateTime() {
                 .orElseThrow(() -> new EntityNotFoundException(Account.class));
 
         CAN_EDIT_ENROLLMENTS.checkAndThrow(STUDY_ID, studyId, USER_ID, account.getId());
```

Comment:
This happens in the above check. Because this was complicated enough that I had to sit and puzzle it out during a code review, I've added some comments to the code explaining what each section is checking (security-wise).